src/proto/single/resp.rs

```diff
@@ -1,8 +1,7 @@
-use crate::FaktoryError;
-use failure::Error;
+use crate::error::{Error, Protocol};
```

Comment:
Same nit here: I'd import `crate::error::{self, Error}` and use `error::Protocol` below.